solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -2056,8 +2056,10 @@ public UpdateHandler getUpdateHandler() {
   private final ArrayDeque<RefCounted<SolrIndexSearcher>> _searchers = new ArrayDeque<>();
   private final ArrayDeque<RefCounted<SolrIndexSearcher>> _realtimeSearchers = new ArrayDeque<>();
 
+  // Single threaded pool, with a time-to-keep of 60 seconds
   final ExecutorService searcherExecutor =
-      ExecutorUtil.newMDCAwareSingleThreadExecutor(new SolrNamedThreadFactory("searcherExecutor"));
+      ExecutorUtil.newMDCAwareSingleLazyThreadExecutor(
+          new SolrNamedThreadFactory("searcherExecutor"));
```

Comment:
minor: mentioning the 60s in the comment shows/assumes `newMDCAwareSingleLazyThreadExecutor` implementation detail, could potentially avoid that (and allow other future use) by adding  arguments to the method/call e.g.

```suggestion
          new SolrNamedThreadFactory("searcherExecutor"), 60L, TimeUnit.SECONDS);
```

Reply:
Thanks for the suggestion. Added parameters for time to keep.